code/modules/mob/living/carbon/human/species_types/zombies.dm

```diff
@@ -24,10 +27,20 @@
 /datum/species/zombie/infectious/spec_stun(mob/living/carbon/human/H,amount)
 	. = min(2, amount)
 
+/datum/species/zombie/infectious/apply_damage(damage, damagetype = BRUTE, def_zone = null, blocked, mob/living/carbon/human/H)
+	. = ..()
+	if(.)
+		regenerating = FALSE
+		addtimer(CALLBACK(src, .proc/Resume_Regeneration), REGENERATION_DELAY, TIMER_UNIQUE|TIMER_OVERRIDE)

```

Comment:
You dont need to do any proc calls or timers

Just regen_cooldown = world.time + REGENERATION_DELAY

And then if(regen_cooldown < world.time)

Reply:
You're the first maintainer to tell me scrap a timer for a world.time cooldown deal.

Not disagreeing but it seemed like it was dicks out for addtimer earlier this year and I'm still not sure when to use what.